(Hopefully) Reverts TG's Cloning Removal and adds Synthflesh cloning as a mid-game tech

SgtHunk opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

This reverts what the Cloning removal did for the most part, and adds Beestation's Synthflesh Cloning. Cloning will now be locked behind research, so engineers along with medical (With the help of chemists) will have to cooperate to build a functional Cloning.

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

Sam wanted this. Along with bringing more ways to get someone back into the game even if they're just a brain. 

## Changelog
:cl:
add: Reverted Cloning Removal
add: Ported Synthflesh cloning from Beestation
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

SgtHunk committed:
Please work

This reverts what the Cloning removal did for the most part, and adds Beestation's Synthflesh Cloning

JohnFulpWillard commented:
speedmerge this man

SgtHunk commented:
Cloning revert speedrun any%

Samuuu3 commented:
Chad

GoldenAlpharex commented:
Just to be sure, it clones your entire body, not just the brain, right?

GoldenAlpharex commented on code/game/machinery/dna_scanner.dm:
```diff
@@ -146,24 +146,3 @@
-
-/obj/item/disk/data/examine(mob/user)
-	. = ..()
-	. += "The write-protect tab is set to [read_only ? "protected" : "unprotected"]."
```
Why exactly are you removing the DNA disks? Are you sure this won't mess with Genetics?

GoldenAlpharex commented on code/game/objects/items/circuitboards/machine_circuitboards.dm:
```diff
@@ -910,8 +910,8 @@
 
-/obj/item/circuitboard/machine/dnascanner
-	name = "DNA Scanner (Machine Board)"
+/obj/item/circuitboard/machine/clonescanner
+	name = "Cloning Scanner (Machine Board)"
```
I'm curious, what's the reason behind renaming this? Did you check to see if it wouldn't tamper with Genetics?

GuillaumePrata replied:
I assume the clone scanner is just a improved version of the dna scanner, still probably better to keep both version or to swap the code from the clone scanner inside the dna scanner.
Having 2 machines still is the easiest way to go IMO.

SgtHunk committed:
Why do geneticists need them anyways

Adds DNA Disks again, introduces Cloning disks. Yes, they're separate. Each work on their corresponding computers.

E231kei500bandai commented:
Cloning was removed from TG because it was bad for roleplay. Why would we remove it on fulpstation, an MRP server. The reason it was removed to begin with was why do anything as as doctor when you can just throw them in cryo if alive, cloning if dead. It just makes doctor incredibly boring and there basically isn't a point. Plus with the TG changes to chem, using chems like libital or aiuri is WORSE than just putting them in cryo or just letting them die in crit to clone them.

SgtHunk commented:
Downstream moment

AdvilPMSevere commented:
Based levels off the charts

SgtHunk committed:
FUCK

Salchems were a mistake
-Fixes cloning requiring an unobtainable/hidden chem, and instead requires the synthflesh libital/carbon/blood recipe

ilikesaying commented:
> Cloning was removed from TG because it was bad for roleplay. Why would we remove it on fulpstation, an MRP server. The reason it was removed to begin with was why do anything as as doctor when you can just throw them in cryo if alive, cloning if dead. It just makes doctor incredibly boring and there basically isn't a point. Plus with the TG changes to chem, using chems like libital or aiuri is WORSE than just putting them in cryo or just letting them die in crit to clone them.

yes but now it'll require science to research cloning, engineers to build the machines and chemistry to make or automate synthflesh, locking it behind TIME

SgtHunk commented:
If Cloning still feels too early, I can lock it behind more research or make it more expensive to research, if that makes you comfortable

E231kei500bandai commented:
fulp moment they test merged it before checking to see if it worked AT ALL like what the fuck

SgtHunk commented:
Blame Salchems, already fixed on last commit

E231kei500bandai commented:
still doesnt work still says Cannot start cloning: Not enough synthflesh. even though there is synthflesh

SgtHunk commented:
Last commit must not be active yet then, I did message a mod and they said that the tgs did not show any updates regarding it, so you're still stuck on the 2nd commit

Kholdgrave commented:
Fantastic. Nice to see that someone took the advice of the majority of players and put cloning back in. I'm legit excited about seeing a more efficient medbay and players not clearing out due to a lack of revival.

SgtHunk commented:
aAaAaaaaaa undefined proc messes up UI, thank you proc very cool. Will fix it ASAP

remove32 commented:
> 
> 
> fulp moment they test merged it before checking to see if it worked AT ALL like what the fuck

what do you think a testmerge is for

ATH1909 commented:
> Cloning was removed from TG because it was bad for roleplay.

https://giphy.com/gifs/just-blinking-3ELtfmA4Apkju

> The reason it was removed to begin with was why do anything as as doctor when you can just throw them in cryo if alive, cloning if dead. It just makes doctor incredibly boring and there basically isn't a point. 

I was gonna respond to that first sentence, but you said what I would have said for me.

> Plus with the TG changes to chem, using chems like libital or aiuri is WORSE than just putting them in cryo or just letting them die in crit to clone them.

Actually, due to repeated misuse by doctors, the organ damage of libital, aiuri, etc. got watered down to the point where it's basically negligible now. The problem was less with cryo and more that cloning was faster, easier, and more multitask-able than literally every other method of reviving people (short of staves of healing and the like). It was at the point where if you spent the time to heal up a corpse and defib it instead of just cloning it, the person who you rezzed would get MAD at you for wasting their time (and yours).

ATH1909 commented on code/datums/brain_damage/phobia.dm:
```diff
@@ -31,6 +31,11 @@
 
+
+/datum/brain_trauma/mild/phobia/on_clone()
+	if(clonable)
```
*clone**a**ble

BootlegBow commented:
putting it behind research really fixes the issue of doctors just cloning instead of something else, although it might make advanced tech like limb growers and advanced tend wounds obsolete

Kholdgrave commented:
Personally, I feel like limb growers and organ harvesters never see any real use. People usually print the Cybernetic organs and go to Robotics for replacement limbs. Advanced Tend Wounds would still see use since even when Cloners were implemented in their original state, there would be piles of corpses that had to wait in a queue, so people can just repair and revive while cloners do their thing. It'll be especially useful to have cloners during Station Wars like Ops or Blob, where the crew can be bled out through sheer attrition since the doctors can't possibly keep up with all the bodies, if the doctors are even alive and medbay is in tact.
